substrate/frame/staking-async/src/pallet/mod.rs

```diff
@@ -2560,5 +2684,44 @@ pub mod pallet {
 
 			Ok(Pays::No.into())
 		}
+
+		/// Perform one step of era pruning to prevent DoS from unbounded deletions.
```

Comment:
Its not a DoS attack we are preventing. We are preventing block weight from exceeding PoV right?